tests/test_dataserver.py

```diff
@@ -1,8 +1,32 @@
+from pathlib import Path
 from hifz.dataserver import DataServer
 from hifz.models import Card
 
+def test_dataserver_returns_card_csv():
+    server = DataServer()
+    temp_file_path = "data/test_arabic.csv"
+    with Path(temp_file_path).open() as f:
+        f.writelines([
+            "front,back",
+            "ا,alif (a as in bad or A as in fAther)",
+            "ب,baa'"
+            ])
+    cards = server.read_entries(temp_file_path)
+    Path(temp_file_path).unlink()
+    assert Card("ب", "baa'") in cards
 
-def test_dataserver_returns_card():
+def test_dataserver_returns_card_json():
     server = DataServer()
-    cards = server.read_entries("data/arabic_letters.csv")
+    temp_file_path = "data/test_arabic.json"
```

Comment:
Use a tmp_path fixture 

[docs](https://docs.pytest.org/en/stable/how-to/tmp_path.html)